__tests__/unit/integration/api.test.js

```diff
@@ -0,0 +1,148 @@
+const { appendToFile } = require("../../../routes/api");
+const fs = require("fs");
+const app = require("../../../app");
+const request = require("supertest");
+
+describe("appendToFile()", () => {
+  test("adds to a file", function () {
+    let testFile = fs.writeFile("./facts-dump/test.txt", "", (err) => {
+      if (err) throw err;
+    });
+    let writeToTestFile = appendToFile("./facts-dump/test.txt", "hello world");
+    fs.readFile("./facts-dump/test.txt", "utf8", (err, data) => {
+      if (err) throw err;
+
+      expect(data).toEqual("hello world");
+    });
+  });
+});
+
+describe("routes", () => {
+  // done() pattern needed for all tests
+  // without done() we get a jest TCPSERVERWRAP error
+
+  // test for route "/:num(-?[0-9]+)"
+
+  test("1st route", async function () {
+    const res = await request(app).get("/1");
+    expect(res.text).toContain("1");
+  });
+
+  // tests for route "/:num([-0-9.,]+)"
+
+  test("batch request with only range", async function () {
+    const res = await request(app).get("/1..3");
+    expect(res.body).toHaveProperty("1", "2", "3");
+  });
+
+  test("batch request with range and specific number", async function () {
+    const res = await request(app).get("/1..3,10");
+    expect(res.body).toHaveProperty("1", "2", "3", "10");
+  });
+
+  test("invalid url for batch request", async function () {
+    const res = await request(app).get("/1...3");
+    expect(res.text).toEqual("Invalid url");
```

Comment:
re: my earlier comment, here's a case where i'd be interested in the status code. is this 400 level or 200 level? i feel like it should be the former, but maybe it's not. either way, making an assertion on the status code will help clarify.